openmls/src/treesync/diff.rs

```diff
@@ -525,32 +573,52 @@ impl<'a> TreeSyncDiff<'a> {
     pub(crate) fn copath_resolutions(
         &self,
         leaf_index: LeafNodeIndex,
-        excluded_indices: &HashSet<&LeafNodeIndex>,
-    ) -> Vec<Vec<HpkePublicKey>> {
+    ) -> Vec<Vec<(TreeNodeIndex, NodeReference)>> {
         // If we're the only node in the tree, there's no copath.
         if self.diff.leaf_count() == 1 {
             return vec![];
         }
 
-        // We want the full path here, including the leaf itself, but not the
-        // root.
-        let mut full_path = vec![TreeNodeIndex::Leaf(leaf_index)];
-        let mut direct_path = direct_path(leaf_index, self.diff.size());
-        if !direct_path.is_empty() {
-            // Remove root
-            direct_path.pop();
+        let mut copath_resolutions = Vec::new();
+        for node_index in self.diff.copath(leaf_index) {
+            let resolution = self.resolution(node_index, &HashSet::new());
+            copath_resolutions.push(resolution);
+        }
+        copath_resolutions
```

Comment:
Not crucial, but since we're using iterators everywhere, we might as well do it here, too. The same goes for the function below.